docs/pages/machine-id/architecture.mdx

```diff
@@ -0,0 +1,46 @@
+---
+title: Machine ID Architecture
+description: How Teleport Machine ID works.
+---
+
+This section provides an overview of Teleport Machine ID's inner workings.
+
+A more in-depth specification of the workings of Machine ID can be found in
+[the Request For Discussion.](https://github.com/gravitational/teleport/blob/master/rfd/0064-bot-for-cert-renewals.md)
+
+## Role Impersonation
```

Comment:
I don't know if we want to include them in this PR or a follow-up, but a couple other sections that might be useful:
* Onboarding, especially the differences between token and IAM joining
* Renewable certificates and the generation counter / detecting stolen certs
* File permissions (or: how Machine ID ensures destinations are only readable by the target application)